app/src/main/java/org/hyperledger/besu/controller/ConsensusScheduleBesuControllerBuilder.java

```diff
@@ -225,9 +227,37 @@ protected JsonRpcMethods createAdditionalJsonRpcMethodFactory(
   protected SubProtocolConfiguration createSubProtocolConfiguration(
       final EthProtocolManager ethProtocolManager,
       final Optional<SnapProtocolManager> maybeSnapProtocolManager) {
-    return besuControllerBuilderSchedule
-        .get(besuControllerBuilderSchedule.keySet().stream().skip(1).findFirst().orElseThrow())
-        .createSubProtocolConfiguration(ethProtocolManager, maybeSnapProtocolManager);
+    // During consensus migration, we need BOTH wire protocols registered.
+    // This allows nodes to communicate before AND after the migration block.
+    // Previously used .skip(1) which was non-deterministic with HashMap ordering.
+    //
+    // We merge all sub-protocol configurations, which registers both:
+    // - IBF/1 (IBFT2) for pre-migration communication
+    // - istanbul/100 (QBFT) for post-migration communication
+    final SubProtocolConfiguration mergedConfig = new SubProtocolConfiguration();
+    final java.util.Set<String> addedProtocolNames = new java.util.HashSet<>();
```

Comment:
nit: be nicer add an import for HashSet so we don't need the fully qualified name